workflow-testing/src/main/java/com/squareup/workflow1/testing/RenderTestResult.kt

```diff
@@ -38,10 +38,59 @@ public interface RenderTestResult<PropsT, StateT, OutputT, RenderingT> {
    * be useful if your actions are anonymous. If they are a sealed class or enum, use [verifyAction]
    * instead and write separate unit tests for your action implementations.
    */
+  @Deprecated(
+    message = "Deprecated in favour of using verify methods with unwrapped output which target " +
+      "either state or output specifically.\n" +
+      "See: verifyActionState, verifyActionOutput, verifyNoActionOutput",
+  )
   public fun verifyActionResult(
     block: (newState: StateT, output: WorkflowOutput<OutputT>?) -> Unit
   ): RenderTestResult<PropsT, StateT, OutputT, RenderingT>
 
+  /**
+   * Asserts that the render pass handled either a workflow/worker output or a rendering event,
+   * "executes" the action with the state passed to [testRender], then invokes [block] with the
+   * resulting state.
+   *
+   * If the workflow didn't process any actions, `newState` will be the initial state.
+   *
+   * Note that by using this method, you're also testing the implementation of your action. This can
```

Comment:
I think the documentation for these 3 methods could be simplified, I took the approach of copy pasting the verifyActionResult docs to each of them since I wasn't sure which part of those docs were really important to maintain.